### ISSUE TITLE
Fatal Protobuf symbol collision during `yolo benchmark` (TensorFlow vs MNN conflict)

### ISSUE THREAD

**Issue by banu4prasad** [ultralytics/ultralytics#24458]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

When running the `yolo benchmark` command in a Google Colab environment, the process terminates with a fatal C++ error. The crash occurs due to a Protocol Buffer symbol collision between TensorFlow and MNN after the Ultralytics AutoUpdate installs MNN dependencies.

The specific error is:
`Symbol name "tensorflow.ResourceHandleProto" conflicts with the existing symbol "tensorflow.ResourceHandleProto"`

Output:

PyTorch: starting from '/content/YOLO26m.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 14, 8400) (42.0 MB)
requirements: Ultralytics requirements ['paddlepaddle-gpu>=3.0.0,<3.3.0', 'x2paddle'] not found, attempting AutoUpdate...
WARNING ⚠️ Retry 1/2 failed: Command 'uv pip install --no-cache-dir --python "/usr/bin/python3" "paddlepaddle-gpu>=3.0.0,<3.3.0" "x2paddle"  --index-strategy=unsafe-best-match --break-system-packages' returned non-zero exit status 1.
WARNING ⚠️ Retry 2/2 failed: Command 'uv pip install --no-cache-dir --python "/usr/bin/python3" "paddlepaddle-gpu>=3.0.0,<3.3.0" "x2paddle"  --index-strategy=unsafe-best-match --break-system-packages' returned non-zero exit status 1.
WARNING ⚠️ requirements: ❌ Command 'uv pip install --no-cache-dir --python "/usr/bin/python3" "paddlepaddle-gpu>=3.0.0,<3.3.0" "x2paddle"  --index-strategy=unsafe-best-match --break-system-packages' returned non-zero exit status 1.
Using Python 3.12.13 environment at: /usr
  × No solution found when resolving dependencies:
  ╰─▶ Because only paddlepaddle-gpu<=2.6.2 is available and you require
      paddlepaddle-gpu>=3.0.0,<3.3.0, we can conclude that your requirements
      are unsatisfiable.

ERROR ❌ PaddlePaddle: export failure 2.4s: No module named 'x2paddle'
ERROR ❌ Benchmark failure for PaddlePaddle: No module named 'x2paddle'
YOLO26m summary (fused): 132 layers, 20,357,162 parameters, 0 gradients, 67.9 GFLOPs

PyTorch: starting from '/content/YOLO26m.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 300, 6) (42.0 MB)

ONNX: starting export with onnx 1.21.0 opset 18...
ONNX: slimming with onnxslim 0.1.93...
ONNX: export success ✅ 4.4s, saved as '/content/YOLO26m.onnx' (78.0 MB)
requirements: Ultralytics requirement ['MNN>=2.9.6'] not found, attempting AutoUpdate...
Using Python 3.12.13 environment at: /usr
Resolved 2 packages in 205ms
Prepared 1 package in 310ms
Installed 1 package in 8ms
 + mnn==3.5.0

requirements: AutoUpdate success ✅ 0.8s
WARNING ⚠️ requirements: Restart runtime or rerun command for updates to take effect

The device supports: i8sdot:0, fp16:0, i8mm: 0, sve2: 0, sme2: 0
E0000 00:00:1778511010.190688    6256 descriptor_database.cc:660] Symbol name "tensorflow.ResourceHandleProto" conflicts with the existing symbol "tensorflow.ResourceHandleProto".
F0000 00:00:1778511010.190759    6256 descriptor.cc:2236] Check failed: GeneratedDatabase()->Add(encoded_file_descriptor, size) 
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
[symbolize_elf.inc : 1422] RAW: /usr/local/lib/python3.12/dist-packages/tensorflow/libtensorflow_framework.so.2 (deleted): open failed: errno=2
*** Check failure stack trace: ***
    @     0x7e1f16a1afe4  (unknown)
    @     0x7e1f16a1a976  (unknown)
    @     0x7e1f16a1b539  (unknown)
    @     0x7e1f1690d5cb  (unknown)
    @     0x7e1f16987308  (unknown)
    @     0x7e1f169872fa  (unknown)
    @     0x7e1f169872fa  (unknown)
    @     0x7e22c6c3947e  (unknown)

### Environment

Ultralytics 8.4.48 🚀 Python-3.12.13 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 49.6/112.6 GB disk)

OS                     Linux-6.6.122+-x86_64-with-glibc2.35
Environment            Colab
Python                 3.12.13
Install                git
Path                   /content/ultralytics/ultralytics
RAM                    12.67 GB
Disk                   49.6/112.6 GB
CPU                    Intel Xeon CPU @ 2.00GHz
CPU count              2
GPU                    Tesla T4, 14913MiB
GPU count              1
CUDA                   12.8

numpy                  ✅ 2.0.2>=1.23.0
matplotlib             ✅ 3.10.0>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.4>=2.23.0
scipy                  ✅ 1.16.3>=1.4.1
torch                  ✅ 2.10.0+cu128>=1.8.0
torch                  ✅ 2.10.0+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0+cu128>=0.9.0
psutil                 ✅ 5.9.5>=5.8.0
polars                 ✅ 1.35.2>=0.20.0
ultralytics-thop       ✅ 2.0.19>=2.0.18

### Minimal Reproducible Example

```
git clone https://github.com/ultralytics/ultralytics -b main
uv pip install -qe ultralytics
```

```
yolo benchmark model=/content/YOLO26m.pt data=/content/dataset.yaml imgsz=640 device=0
```

### Additional

is there any way to get benchmark of only export these formats

Pytorch, ONNX, TensortRT

i meant 
| Format | Status | Size (MB) | metrics/mAP50-95(B) | Inference time (ms/im) | FPS |
| :--- | :---: | :---: | :---: | :---: | :---: |
| PyTorch | ✅ | 41.6 | 0.2311 | 19.57 | 51.09 |
| ONNX | ✅ | 77.2 | 0.2309 | 21.72 | 46.05 |
| TensorRT | ✅ | 86.2 | 0.2309 | 20.17 | 49.58 |

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @banu4prasad, thank you for the detailed bug report and for sharing a reproducible example 🚀 This is an automated response to help move things along, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this is a 🐛 Bug Report, thanks for already including a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) 🙌

To help with investigation, please also confirm:
- Whether the crash still occurs after upgrading to the latest `ultralytics` package
- Whether this happens consistently in a fresh Colab runtime after a runtime restart
- The exact command or workflow if you are trying to benchmark only a subset of export formats
- Any additional logs from a clean environment if available

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by AffanBinFaisal** [comment]
**What's going wrong**

- **Crash:** TensorFlow (common on Colab) and **MNN** both register the same Protobuf types in one Python process. That triggers a **fatal native error**, not a normal Python exception.
- **Paddle errors in the log:** Those are **failed Paddle exports** only. They are **not** what kills the process.

**Workaround**
`benchmark()` can run **one** format at a time via `format=` in `ultralytics/utils/benchmarks.py`. Call it **once per format** you care about so MNN (and Paddle) are never run.

**Possible fixes (for maintainers)**

1. **Skip MNN** when TensorFlow is already available, with a clear log message.
2. Or run **MNN export in a subprocess** so it doesn't share Protobuf state with TF.

**Comment by banu4prasad** [comment]
```
yolo benchmark ......
```

CLI cmd is working on Kaggle but not on colab

Format | Status | Size (MB) | metrics/mAP50-95(B) | Inference time (ms/im) | FPS
-- | -- | -- | -- | -- | --
PyTorch | ✅ | 14.5 | 0.21 | 17.32 | 57.74
TorchScript | ✅ | 28.4 | 0.2092 | 10.53 | 94.92
ONNX | ✅ | 27.7 | 0.2092 | 10.03 | 99.65
OpenVINO | ❌ | 0 | - | - | -
TensorRT | ✅ | 38.4 | 0.2092 | 8.0 | 124.98
CoreML | ❌ | 0 | - | - | -
TensorFlow SavedModel | ✅ | 69.4 | 0.2092 | 22.2 | 45.03
TensorFlow GraphDef | ✅ | 27.7 | 0.2092 | 21.4 | 46.72
TensorFlow Lite | ❌ | 0 | - | - | -
TensorFlow Edge TPU | ❌ | 0 | - | - | -
TensorFlow.js | ❌ | 0 | - | - | -
PaddlePaddle | ❌ | 0 | - | - | -
MNN | ✅ | 27.5 | 0.2092 | 196.79 | 5.08
NCNN | ✅ | 27.6 | 0.2092 | 166.21 | 6.02
IMX | ❌ | 0 | - | - | -
RKNN | ❌ | 0 | - | - | -
ExecuTorch | ❌ | 0 | - | - | -
Axelera AI | ❌ | 0 | - | - | -

**Comment by Y-T-G** [comment]
To use benchmark, it's recommended to install Ultralytics with `export` option to avoid dependency issues.


```
pip install ultralytics[export]
```

**Comment by banu4prasad** [comment]
@Y-T-G should i close this issue?

**Comment by Y-T-G** [comment]
You can close it if it's resolved
